oak_functions_service/src/lookup.rs

```diff
@@ -161,7 +161,7 @@ impl LookupDataManager {
     pub fn create_lookup_data(&self) -> LookupData {
         let keys;
         let data = {
-            let data = self.data.lock().clone();
+            let data = self.data.read().clone();
```

Comment:
This is surprising, since I would not have expected cloning of an Arc to take any signigicant amount of time. Perhaps the kernel occasionally switched away from a thread while it held the lock, blocking all other threads until it continued executing again. If that is the case, we can perhaps get better performance by using `parking_lot` for Mutexes on environments that have a std lib. `lock_api` is used by both, so it should be possible to use a single abstration. For the RW lock with rare writes I don't think it will make a lot of difference, but might be worth investigating as a future improvement for other locks.

Reply:
I think it was just lock contention as we're dealing with spinlocks here. Yes, cloning an Arc doesn't take that long, but I think internally `Spinlock` has an exponential backoff when trying to grab the lock, so the lock can easily take much longer to grab the lock than the underlying operation would imply.

Reply:
I think there is a separate BackoffSpinlock class for exponential backoff, which we are not using, so this is still a very surprising finding.